### PR TITLE
Add e2e test for Deployment controllerRef orphaning and adoption

### PR DESCRIPTION
**Pull request by janetkuo**
Follow up #42908 

@enisoc @kubernetes/sig-apps-bugs @kargakis

### CODE REVIEW

**Comment by janetkuo** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot unit test this

**Comment by janetkuo** [comment]
@k8s-bot kops aws e2e test this

**Comment by janetkuo** [comment]
@k8s-bot kops aws e2e test this

File: test/e2e/deployment.go [hunk 1]
```diff
@@ -1289,3 +1297,91 @@ func replicaSetHasDesiredReplicas(rsClient extensionsclient.ReplicaSetsGetter, r
...
+	Expect(err).NotTo(HaveOccurred())
+	err = framework.WaitForDeploymentStatus(c, deploy)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Checking its ReplicaSet has the right controllerRef")
+	err = checkDeploymentReplicaSetsControllerRef(c, ns, deploy.UID, podLabels)
+	Expect(err).NotTo(HaveOccurred())
+
+	By(fmt.Sprintf("Deleting Deployment %q and orphaning its ReplicaSets", deploymentName))
+	err = orphanDeploymentReplicaSets(c, deploy)
```
Comment (kargakis) [inline test/e2e/deployment.go]: missing error check

Comment (janetkuo) [inline test/e2e/deployment.go]: Oops, fixed

**Comment by janetkuo** [comment]
@k8s-bot kops aws e2e test this

**Pull request merged**